rclcpp/src/rclcpp/node.cpp

```diff
@@ -236,34 +256,56 @@ Node::get_parameter(const std::string & name) const
   return node_parameters_->get_parameter(name);
 }
 
-bool Node::get_parameter(
-  const std::string & name,
-  rclcpp::Parameter & parameter) const
+bool
+Node::get_parameter(const std::string & name, rclcpp::Parameter & parameter) const
 {
   return node_parameters_->get_parameter(name, parameter);
 }
 
-std::vector<rcl_interfaces::msg::ParameterDescriptor>
-Node::describe_parameters(
+std::vector<rclcpp::Parameter>
+Node::get_parameters(
   const std::vector<std::string> & names) const
+{
+  return node_parameters_->get_parameters(names);
+}
+
+rcl_interfaces::msg::ParameterDescriptor
+Node::describe_parameter(const std::string & name) const
+{
+  auto result = node_parameters_->describe_parameters({name});
+  if (0 == result.size()) {
+    throw rclcpp::exceptions::ParameterNotDeclaredException(name);
+  }
+  if (result.size() > 1) {
+    throw std::runtime_error("number of described parameters unexpectedly more than one");
+  }
+  return result.front();
+}
+
+std::vector<rcl_interfaces::msg::ParameterDescriptor>
+Node::describe_parameters(const std::vector<std::string> & names) const
 {
   return node_parameters_->describe_parameters(names);
 }
 
 std::vector<uint8_t>
```

Comment:
@wjwwood would it make sense to return an `std::vector<rclcpp::ParameterType>` instead?

Reply:
Maybe, I didn't change it because that is how it was before, and I thought it might have something to do with the enum existing in two places (once in `rclcpp` and once as constants in `rcl_interfaces::msg::ParameterType`. I think I'll leave it for now?